lib/search/models/search_result.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:equatable/equatable.dart';
+
+enum SearchResultType { session, speaker, sponsor, organizer }
+
+class SearchResult extends Equatable {
+  final String id;
+  final String title;
+  final String subtitle;
+  final String imageUrl;
+  final SearchResultType type;
+  final dynamic extra;
```

Comment:
From your approach, there's a way we can avoid `dynamic` data. You have `type` that you assign in the cubit with the search results. As such, you can have shave `extra` and then have a nullable type for `Session`, `Speaker`, ...etc. Then based on your current implementation in the UI, when the type matches, example
```
case SearchResultType.speaker:
```

Speaker? speaker wouldn't be null and you can use that.

Reply:
This has been fixed